### PR TITLE
Help button workflow 

### LINKED ISSUE #1612: Integrate Help button use case in catalog app.
**Is your feature request related to a problem? Please describe.**
As Help button feature will be part of sdc module, it will be good to integrate same in any one of the existing screen of the catalog app.

@shelaghm Can you please suggest use case in catalog app where help button can be shown.

#1528

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Would you like to work on the issue?**
Please state if this issue should be assigned to you or who you think could help to solve this issue.

### PR DESCRIPTION
**Pull request by santosh-pingle**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1612 

**Description**
- Show help button workflow use case.

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Feature

**Screenshots (if applicable)**
Screen Shot 2022-11-03 at 11 33 46 AM

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: catalog/src/main/assets/component_help.json [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            }
+          ],
+          "linkId": "1.3",
+          "text": "Only one answer can be selected. If none apply, skip the question.",
+          "type": "display"
+        }
+      ]
+    }
+  ]
+}
```
Comment (omarismail94) [inline catalog/src/main/assets/component_help.json:79]: nit: add empty line at the bottom

**Review by omarismail94 — APPROVED** [review]
Nice change @santosh-pingle ! Just left one nit

**Comment by santosh-pingle** [comment]
@shelaghm 
Screen Shot 2022-11-21 at 7 03 17 PM
Screen Shot 2022-11-21 at 7 03 40 PM

**Comment by shelaghm** [comment]
@santosh-pingle looks good. Thanks!

File: catalog/src/main/java/com/google/android/fhir/catalog/ComponentListViewModel.kt [hunk 2]
```diff
@@ -30,89 +33,98 @@ class ComponentListViewModel(application: Application, private val state: SavedS
...
-    val questionnaireFile: String,
+    val questionnaireFile: String = "",
     /**
      * Path to the questionnaire json file with some or all required fields. If the user doesn't
      * answer the required questions, an error may be displayed on the particular question.
      */
     val questionnaireFileWithValidation: String = "",
     val workflow: WorkflowType = WorkflowType.COMPONENT
   ) {
+    WIDGETS(textId = R.string.widgets, viewType = HEADER_TYPE),
```
[the remark this answers is not in the document]
Comment (jingtang10) [reply catalog/src/main/java/com/google/android/fhir/catalog/ComponentListViewModel.kt:48]: you can just create another list that is used as the order for the grid.

**Review by jingtang10 — CHANGES_REQUESTED** [review]

File: catalog/src/main/java/com/google/android/fhir/catalog/ComponentListViewModel.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (santosh-pingle) [reply catalog/src/main/java/com/google/android/fhir/catalog/ComponentListViewModel.kt:48]: looking into it.

[the remark this answers is not in the document]
Comment (santosh-pingle) [reply catalog/src/main/java/com/google/android/fhir/catalog/ComponentListViewModel.kt]: done

File: catalog/src/main/java/com/google/android/fhir/catalog/ComponentListViewModel.kt [hunk 3]
```diff
@@ -114,5 +121,25 @@ class ComponentListViewModel(application: Application, private val state: SavedS
       R.string.component_name_repeated_group,
       "component_repeated_group.json",
     ),
+    HELP(R.drawable.ic_help, R.string.component_name_help, "component_help.json")
   }
+
+  val viewItemList =
+    listOf(
+      ViewItem.HeaderItem(Header.WIDGETS),
```
Comment (jingtang10) [inline catalog/src/main/java/com/google/android/fhir/catalog/ComponentListViewModel.kt:129]: optional:

if you add a function in the `Header` enum and the `Component` enum called something like `toViewItem` and wraps the enum value in a `ViewItem` I think that might make this list looks a little more succint:

```
val viewItemList =
    listOf(
      Header.WIDGETS.toViewItem(),
      Component.BOOLEAN_CHOICE.toViewItem(),
      ...
```

up to you though

**Review by jingtang10 — APPROVED** [review]
thanks @santosh-pingle for this change! looks much cleaner!

File: catalog/src/main/java/com/google/android/fhir/catalog/ComponentListViewModel.kt (hunk 3 above)
Comment (jingtang10) [reply catalog/src/main/java/com/google/android/fhir/catalog/ComponentListViewModel.kt:129]: i've done this.

**Pull request merged**